src/engraving/rendering/dev/measurelayout.cpp

```diff
@@ -1173,10 +1173,47 @@ void MeasureLayout::layoutMMRestRange(Measure* m, LayoutContext& ctx)
         }
         // setXmlText is reimplemented to take care of brackets
         rr->setXmlText(s);
-        TLayout::layoutMMRestRange(rr, rr->mutldata());
+        TLayout::layoutMMRestRange(rr, rr->mutldata(), ctx);
     }
 }
 
+std::pair<double, double> MeasureLayout::getMeasureStartEndPos(const Measure* measure, const Segment* firstCrSeg,
```

Comment:
Couple of things: 
- I'm not a fan of using pairs for returning two values (a pair to me is almost exclusively associated with maps, whose elements are key-value pairs). I normally either create a struct to hold them, or make the function void and pass them in as references.
- It seems to me that this function is only really used in one place, where we then take the mid position of x1 and x2. At that point, I think it would be more elegant to call this function getMeasureVisualCenter or something like that and return the center point directly, rather than returning the two values

Reply:
Agreed on pairs, I'll use a struct.  
The reason I opted to return two values instead of the centre of the bar is it's use in `MeasureLayout::layoutMeasureElements` where the values aren't always used to calculate a centre point (MMRests, measure repeats, measure rests).  I'll try to get the same points just from the centre position - I agree it'll tidy things up a lot.

Reply:
No no, if there _is_ a use for x1 and x2 that's different from just finding the center, then by all means let's have the function return x1 and x2. It just seemed to me that it was only used for the center but I might have just missed it

Reply:
In other words, if the only use for x1 and x2 is to do `center = 0.5 * (x1 + x2)` then it makes more sense for this function to just return the center directly. But if x1 and x2 are used for other stuff, then it's fine as it is (but without using the pair for return)